### ISSUE TITLE
ROS 2 Rolling build: the extension doesn't run the same tests as colcon-python-setup-py

### ISSUE THREAD

**Issue by clalancette** [colcon/colcon-python-project#2]
I've been testing the `devel` branch of this extension out when building and testing with ROS 2 Rolling.

When I try to run tests on `ament_python` packages, it looks like no tests are getting run with this extension, while with `colcon-python-setup-py`, they are.

The easiest way to reproduce what I'm seeing here is to clone a ROS 2 Rolling workspace using the instructions in http://docs.ros.org/en/rolling/Installation/Alternatives/Ubuntu-Development-Setup.html .  Then in the workspace run:

```
colcon build --event-handlers console_direct+ --packages-up-to ament_pycodestyle
colcon test --event-handlers console_direct+ --packages-select ament_pycodestyle
```

When using `colcon-python-setup-py`, this is the result I see:
```
Starting >>> ament_pycodestyle
============================= test session starts ==============================
platform linux -- Python 3.10.6, pytest-6.2.5, py-1.10.0, pluggy-0.13.0
cachedir: /home/ubuntu/new_ros2_ws/build/ament_pycodestyle/.pytest_cache
rootdir: /home/ubuntu/new_ros2_ws/src/ament/ament_lint/ament_pycodestyle, configfile: pytest.ini
plugins: colcon-core-0.12.1, repeat-0.9.1, mock-3.6.1, timeout-2.1.0, rerunfailures-10.2, cov-3.0.0
collecting ... 
collected 1 item                                                               

test/test_ignore_config.py .                                             [100%]

- generated xml file: /home/ubuntu/new_ros2_ws/build/ament_pycodestyle/pytest.xml -
============================== 1 passed in 0.02s ===============================
Finished <<< ament_pycodestyle [0.41s]          

Summary: 1 package finished [0.98s]
```

And that is exactly what I expect; there is exactly one test in https://github.com/ament/ament_lint/tree/f10eecd28cdc05746c1f8c0f45984db6052cfd40/ament_pycodestyle/test .

However, when running with this extension sourced, I instead see:

```
Starting >>> ament_pycodestyle

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
--- stderr: ament_pycodestyle

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
---
Finished <<< ament_pycodestyle [0.04s]

Summary: 1 package finished [0.77s]
  1 package had stderr output: ament_pycodestyle
```

Let me know if you need additional information.

**Comment by cottsay** [comment]
This is related to the  colcon/colcon-core#548 saga.

The change that PR reverts intended to move from setuptools-specific dependencies to those detected by colcon, but what I failed to realize is that ROS packages have rosdep keys for dependencies, not python package names, so it broke things for ROS and we reverted it.

Now we're on the other side of the problem. We have a non-setuptools Python package so it isn't populating the setuptools-specific metadata on the package descriptor. For pure-python packages, we could additionally inspect the colcon dependencies, but that doesn't help us for `ros.ament_python.project` packages.

The best approach I can think of is to "fake" the setuptools-based dependency information that `colcon-core`'s test dependency detection is expecting.

To work around this issue, you can specify `--python-testing pytest` to your `colcon test` invocation.

**Comment by cottsay** [comment]
Should be solved by [5abc103](https://github.com/colcon/colcon-python-project/pull/1/commits/5abc1032219b89f3c062ef9687c85229cdf39980). Please pull `devel` and give it a shot.

**Comment by clalancette** [comment]
I think this is fixed now, so I'm going to close it out.
